onnxruntime/core/providers/cpu/quantization/conv_integer.cc

```diff
@@ -34,17 +34,18 @@ ONNX_OPERATOR_KERNEL_EX(
     ConvInteger);
 
 Status ConvInteger::Compute(OpKernelContext* context) const {
-  size_t num_inputs = OpKernel::Node().InputDefs().size();
+  const auto input_defs = Node().InputDefs();
+  size_t num_inputs = input_defs.size();
   const auto* X = context->Input<Tensor>(0);
   const auto* W = context->Input<Tensor>(1);
   uint8_t input_offset = 0;
   uint8_t filter_offset = 0;
-  if (num_inputs >= 3) {
+  if (num_inputs >= 3 && input_defs[2]->Exists()) {
```

Comment:
I didn't know this will cause crash... Is the crash happening inside `context->Input<Tensor>(2)` here? I thought in this case it's just `X_Zero_Point` will be `nullptr`.

Reply:
if we get a name for an input the input arg exists (num_inputs is based on this).
if the name is an empty string it's a missing optional input. requires the call to `Exists()`. does NOT only occur at the end of inputs
if the optional input is missing there is no input data, so `context->Input<Tensor>(i)` returns nullptr.

Complicated by ONNX not requiring trailing optional args to be provided, so the input count for an operator like this could be 2, 3 or 4 depending on whether the model specifies "" for the optional inputs or not.

Basically for any optional input you MUST first check if it's present in the inputs args, and MUST check if it exists.

Reply:
X_Zero_Point will be nullptr. IsScalarOr1ElementVector doesn't handle that. Nor should it unless the name changes to indicate it also checks if the Tensor exists. 

Side note: it should take a reference though to clearly advertise that a Tensor instance is required, but that simply moves where it breaks.

https://github.com/microsoft/onnxruntime/blob/2ba076aa9ea8e3c68e97dc0f2463fb2786fcb96f/onnxruntime/core/providers/common.h#L40-L47

Reply:
Thank you for the detailed explanation!